### PR TITLE
Hw1

### PR DESCRIPTION
**Pull request by IVovchynskyi**
Hi, @mmykhailenko  !
The first home work is ready.
There are some additional comments, just to make the final result more readable.

### CODE REVIEW

File: HW1.py [hunk 1]
```diff
@@ -15,4 +15,118 @@
 4) Сгенерировать dict() из списка ключей ниже по формуле (key : key* key).
 keys = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
 ожидаемый результат: {1: 1, 2: 4, 3: 9 …}
-"""
\ No newline at end of file
+"""
+
+# 1.
+InitialArray = [10, 11, 2, 3, 5, 8, 23, 11, 2, 5, 76, 43, 2, 32, 76, 3, 10, 0, 1]
```
Comment (mmykhailenko) [inline HW1.py]: all variables should be lower case

Comment (mmykhailenko) [reply HW1.py]: and with underscore `initial_array`.

File: HW1.py [hunk 2]
```diff
@@ -15,4 +15,118 @@
...
+# 1.1) убрать из него повторяющиеся элементы
+NoDuplicateArray = list(set(InitialArray))
+print("1.1 No duplicates array")
+print(NoDuplicateArray)
+
+# The function to check that we were right
+def FindDuplicateInArray(TheArray):
```
Comment (mmykhailenko) [inline HW1.py]: same rules for name of functions `def find_duplicate_in_array(the_array)`

File: HW1.py [hunk 3]
```diff
@@ -15,4 +15,118 @@
...
+    while bool(TheArray) is True:
```
Comment (mmykhailenko) [inline HW1.py]: `while bool(TheArray) is True:` same as `while the_array:`

File: HW1.py [hunk 4]
```diff
@@ -15,4 +15,118 @@
...
+def NMaxElents(HowManyElements, TheArray):
+    MaxNumbersList = []
+    NoDuplicateArray = list(set(TheArray))  # to avoid the "76, 76, 43" answer
+    while bool(HowManyElements) is True:
+        HowManyElements -= 1
+        MaxNumbersList.append(max(NoDuplicateArray))
+        NoDuplicateArray.remove(max(NoDuplicateArray))
+    return(print(MaxNumbersList))
+print("1.2 Three max values of the array")
+NMaxElents(3, InitialArray)
```
Comment (mmykhailenko) [inline HW1.py]: could be shorted 
```
sorted_list = sorted(the_array, reversed=True)
print(sorted_list[:3])
```

File: HW1.py [hunk 5]
```diff
@@ -15,4 +15,118 @@
...
+        TargetElementQty -= 1
+        MinIndex = TheArray.index(MinValueInArray)
+        MinIndexList.append(MinIndex)
+        0) we cannot just `TheArray.remove(MinValueInArray)` or `TheArray.pop(MinIndex)`
+        because indexes after the first one will be shifted (#2 by 1, #3 by 2, et.c.) 
+        1) So we need to make a correction fro this shift
+        2) OR to replace already found positions by unique values (ideally not from TheArray, but at least any != MinValueInArray)
+        """
+print("1.3 The index of the smallest value of the array")
+ShowMinIndexOfArray(InitialArray)
```
Comment (mmykhailenko) [inline HW1.py]: could be shorted 
`print(the_array.index(min(the_array)))`

File: HW1.py [hunk 6]
```diff
@@ -15,4 +15,118 @@
...
+#2) Сгенерировать массив(list()). Из диапазона чисел от 0 до 100 записать в результирующий массив только четные числа.
+def EvenArrayGenerator(Start, Finish):
+    EvenArray = []
+    for i in range(Start, Finish + 1, 1):
+        if i % 2 == 0:
+            EvenArray.append(i)
+    return(EvenArray)
+EvenArrayGenerator(0,100)
+print("2. Only even array from 0 to 100")
+print(EvenArrayGenerator(0,100))
```
Comment (mmykhailenko) [inline HW1.py]: could be shorted `[i for i in range(100) if i % 2 == 0]`

File: HW1.py [hunk 7]
```diff
@@ -15,4 +15,118 @@
...
+keys = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+def SquareDictFromKeysList(keys):
+    values = []
+    for key in keys:
+        value = key ** 2
+        values.append(value)
+    SquareDict = dict(zip(keys, values))
+    return(SquareDict)
+print("4. Dict with value = key ** 2")
+print(SquareDictFromKeysList(keys))
```
Comment (mmykhailenko) [inline HW1.py]: could be shorted `{ i: i*i for t in keys }`

**Review by mmykhailenko — APPROVED** [review]
Try to find a more easy way for resolving the tasks. Less code -less bugs

File: HW1.py (hunk 4 above)
Comment (IVovchynskyi) [reply HW1.py]: There is a typo: should be without 'd': reverse=True

File: HW1.py (hunk 7 above)
Comment (IVovchynskyi) [reply HW1.py]: "i" is not defined; it should be taken from somewhere; from "keys" for example.
So it should be: `{i: i*i for i in keys}`
With "i" as iterator it works.

**Pull request closed without merging**